src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/EventBookings.java

```diff
@@ -30,6 +30,24 @@
  */
 public interface EventBookings {
 
+    /**
+     * Add booking to the database.
+     *
+     * @param eventId
+     *            - the event id
+     * @param userId
+     *            - the user id
+     * @param reservedById
+     *            - the user id of who made the reservation (can be null)
+     * @param status
+     *            - the initial status of the booking.
+     * @param additionalInformation - additional information required for the event.
+     * @return the newly created booking
+     * @throws SegueDatabaseException
+     *             - if an error occurs.
+     */
+    EventBooking add(final String eventId, final Long userId, final Long reservedById, final BookingStatus status, final Map<String,String> additionalInformation) throws SegueDatabaseException;
+
```

Comment:
Could the add interface method below just be extended with some nullable reservedById parameter? It is perfectly possible for an event manager to make a booking on behalf of someone else and that could be captured using the same parameter. I realise it means having to add the missing parameter to every invocation but there shouldn't be many.

Just a suggestion though. Not crucial - I see you did the nice thing in the actual implementation.

Reply:
The problem is with the slightly twisted semantics. It captures the workflow without actually capturing the workflow. Right now, the presence of that field in the record indicates that there was a reservation before there was a booking. To apply the same meaning to an evento manager booking someone else, we'd have to introduce a `bookedById` rathern than a `reservedById` field.